ffi/src/ffi_tracing.rs

```diff
@@ -312,43 +377,179 @@ fn setup_log_line_subscriber(
     // This repeats some code, but avoids some insane generic wrangling if we try to abstract the
     // type of `fmt_layer` over the formatter
     macro_rules! setup_subscriber {
-        ( $format_fn:ident ) => {
-            if with_time {
-                let fmt_layer = fmt_layer.$format_fn().with_filter(filter);
-                let subscriber = Registry::default().with(fmt_layer).with(tracking_layer);
-                tracing::subscriber::set_global_default(subscriber)
-                    .expect("Failed to set global subscriber");
-            } else {
-                let fmt_layer = fmt_layer.without_time().$format_fn().with_filter(filter);
-                let subscriber = Registry::default().with(fmt_layer).with(tracking_layer);
-                tracing::subscriber::set_global_default(subscriber)
-                    .expect("Failed to set global subscriber");
-            }
-        };
+        ($($transform:ident()).*) => {{
```

Comment:
oh clever -- I didn't realize we could capture `()` and `.` like that!

(only works for these no-arg invocations tho...)

Reply:
Yeah, macros are pretty nice. You can handle arg cases too with nested and repeated args but the macro will get really ugly. I initially wrote a pretty complex macro because I was using the `with_ansi(ansi)` call, and then realized you can simplify a lot if you get rid of the arg and the need to have a call at all.